pkg/controller/jobframework/jobframework_test.go

```diff
@@ -0,0 +1,175 @@
+package jobframework
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	kubeflow "github.com/kubeflow/mpi-operator/pkg/apis/kubeflow/v2beta1"
+	kftraining "github.com/kubeflow/training-operator/pkg/apis/kubeflow.org/v1"
+	rayjobapi "github.com/ray-project/kuberay/ray-operator/apis/ray/v1alpha1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlmgr "sigs.k8s.io/controller-runtime/pkg/manager"
+	jobset "sigs.k8s.io/jobset/api/jobset/v1alpha2"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/util/slices"
+	utiltesting "sigs.k8s.io/kueue/pkg/util/testing"
+)
+
+func TestSetupControllers(t *testing.T) {
+	var errModifyOpts = errors.New("test error")
+
+	cases := map[string]struct {
+		opts       []Option
+		modifyOpts modifyOptions
+		mapperGVKs []schema.GroupVersionKind
+		wantError  error
+	}{
+		"setup controllers succeed": {
+			opts: []Option{
+				WithEnabledFrameworks(sets.New("batch/job", "kubeflow.org/mpijob")),
+			},
+			mapperGVKs: []schema.GroupVersionKind{
+				batchv1.SchemeGroupVersion.WithKind("Job"),
+				kubeflow.SchemeGroupVersionKind,
+			},
+			modifyOpts: func(fwkName string, opts ...Option) ([]Option, error) { return opts, nil },
+		},
+		"modifyOptions returns errors": {
```

Comment:
can we add a test case when kubeflow is enabled, but there is no mapper for it?

Reply:
Proving no errors occur would be valuable In that case. Thanks!

https://github.com/kubernetes-sigs/kueue/blob/5d177736896c2bf56eb4dfdb0ce437f30d73e53d/pkg/controller/jobframework/jobframework.go#L52

Reply:
Done.